### DISCUSSION TITLE
Immer + Persist Middleware Problems

### DISCUSSION THREAD

**Discussion by BassWow** [pmndrs/zustand#1143] [Q&A] [answered]
Hi guys,

I am struggling to under why I keep getting fetchBears  is not a function in my sandbox examples.

I want to use the FLUX method but not sure I am understanding it 100%. I think it's best to create slices similar to redux/toolkit. However I found when I move the async action 'fetchBears' into the root of the state it works, but when I move the action back into the slice it does not work and I keep getting that error. I have two sandboxes demonstrating the issue. Currently using localStorage default.

**Not working**
https://codesandbox.io/s/zustand-immer-zjvk0z?file=/src/store.js

**Working but action out of slice**
https://codesandbox.io/s/practical-wildflower-b3ebfu?file=/src/store.js

**Comment by dai-shi** [comment]
Hmm, looks like a bug in `persist`? If we don't use the persist middleware, it works as expected.
What do you think? @AnatoleLucet

**Comment by AnatoleLucet** [comment, marked as answer]
You need to deep merge the persisted state. See #652

Here's a working example with Ramda's deep merge function: https://codesandbox.io/s/zustand-immer-forked-7x897z?file=/src/store.js

**Reply by dai-shi** [reply]
Ah, I was blind. Totally right.

**Reply by BassWow** [reply]
Thank you so much guys I really appreciate it I had a feeling it has something to do with merging but I just wasn't sure I really appreciate the time you both have taken to answer this!!
